ksqldb-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java

```diff
@@ -274,10 +274,22 @@ public static SqlType fromValue(final BigDecimal value) {
         ? value.setScale(0, BigDecimal.ROUND_UNNECESSARY)
         : value;
 
-    /* When a BigDecimal has value 0, the built-in method precision() always returns 1. To account
-    for this edge case, we just take the scale and add one and use that for the precision instead.
-    */
-    if (decimal.compareTo(BigDecimal.ZERO) == 0) {
+    /* We can't use BigDecimal.precision() directly for all cases, since it defines
```

Comment:
Great documentation and explanation! If we ever decide to strictly follow the SQL standard for precision, it will be clear where to start from